src/mcrypt/ctr.c

```diff
@@ -93,14 +93,20 @@ GF_Err _mcrypt_set_state(void *_buf, void *IV, int size)
 
 GF_Err _mcrypt_get_state(void *buf, void *IV, int *size)
 {
-	if (*size < ((CTR_BUFFER* )buf)->blocksize + 1) {
-		*size = ((CTR_BUFFER* )buf)->blocksize + 1;
+	if (*size < ((CTR_BUFFER* )buf)->blocksize) { 
+		*size = ((CTR_BUFFER* )buf)->blocksize;
 		return GF_BAD_PARAM;
 	}
-	*size = ((CTR_BUFFER* )buf)->blocksize + 1;
```

Comment:
For ctr, size in _mcrypt_get_state (and also _mcrypt_set_state) should be blocksize + 1: first byte is the current counter position; following by blocksize of IV